k9mail/src/main/java/com/fsck/k9/crypto/AutocryptOperations.java

```diff
@@ -0,0 +1,150 @@
+package com.fsck.k9.crypto;
+
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Map;
+
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.VisibleForTesting;
+
+import com.fsck.k9.mail.internet.MimeMessage;
+import com.fsck.k9.mail.internet.MimeUtility;
+import okio.ByteString;
+import org.openintents.openpgp.AutocryptPeerUpdate;
+import org.openintents.openpgp.util.OpenPgpApi;
+import timber.log.Timber;
+
+
+public class AutocryptOperations {
```

Comment:
It feels like this class is doing too many things. Maybe extract the Autocrypt header parser to a separate class. Then the logic to find a valid Autocrypt header in an email to another class. And finally the method that adds data to an OpenPgpApi Intent can be in its own class.